### PR TITLE
[FUCK] Fixes a Few Oversights With the TGUI Latejoin Menu!

### PR DESCRIPTION
**Pull request by RimiNosha**
## About The Pull Request

I fucked up, and it was realized after it was merged!

Fixes:
- No shuttle status
- No round timer
- The UI staying open after spawning (Nevermind, I can't figure out the reason, and need help with this one...)
- The UI spamming the server with act calls while it's open
## Why It's Good For The Game

Less bug good

image

## Changelog
:cl: RimiNosha
fix: Fixed some issues with the TGUI latejoin menu, and made the shuttle and round time display once more.
/:cl:

### CODE REVIEW

**Comment by RimiNosha** [comment]
Need help with the menu staying open after spawn. It broke somewhat recently, as I didn't have this issue before I merged master into the test environment, it seems.

**Comment by JohnFulpWillard** [comment]
Do
```
/datum/latejoin_menu/ui_state(mob/user)
	return GLOB.new_player_state
```
ui_state is what handles whether or not a UI is interactive or should be open at all, GLOB.new_player_state will use ``/datum/ui_state/new_player_state``, which closes if you're not a new player.

**Comment by RimiNosha** [comment]
Nope, doesn't work, something is fuck.
image

File: tgui/packages/tgui/interfaces/JobSelection.tsx [hunk 1]
```diff
@@ -106,16 +106,30 @@ export const JobSelection = (props, context) => {
   );
 
   return (
-    <Window width={1012} height={666 /* Hahahahahaha */}>
+    <Window
+      width={1012}
+      height={data.shuttle_status ? 690 : 666 /* Hahahahahaha */}
+      onComponentDidMount={
+        // Send a heartbeat back to DM to let it know the window is alive and well
+        act('ui_mounted_with_no_bluescreen')
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/JobSelection.tsx:114]: You forgot to wrap it in a function.

Comment (RimiNosha) [reply tgui/packages/tgui/interfaces/JobSelection.tsx:114]: Pain.

**Review by stylemistake — APPROVED** [review]

File: code/modules/mob/dead/new_player/latejoin_menu.dm [hunk 2]
```diff
@@ -161,6 +158,7 @@ GLOBAL_DATUM_INIT(latejoin_menu, /datum/latejoin_menu, new)
 					tgui_alert(owner, "The server is full!", "Oh No!")
 					return TRUE
 
+			SStgui.close_user_uis(owner, src) // Bandaid fix cause ui_state doesn't react properly to spawning in.
```
Comment (Watermelon914) [inline code/modules/mob/dead/new_player/latejoin_menu.dm:161]: remove this, it's redundant with your GLOB.new_player_state state

Comment (Watermelon914) [reply code/modules/mob/dead/new_player/latejoin_menu.dm:161]: nvm this is already merged

**Pull request merged**